### PR TITLE
release(2021-01-20): bump package versions

### PR DESCRIPTION
**Pull request by timtay-microsoft**
## Java IotHub Service Client (com.microsoft.azure.sdk.iot:iot-service-client:1.27.0)
- Add support for providing custom SSLContext to service client instances (#1044)
- Increase dependency versions for commons-codec, jackson (#1039)

## Java IotHub Device Client (com.microsoft.azure.sdk.iot:iot-device-client:1.29.0)
- Add new MultiplexingClient that allows for dynamic adding and removal of devices from a multiplexed connection even after opening the connection, deprecate previous multiplexing client TransportClient (#939)

### Bug fixes
- Fix cleanup logic in AMQP layer when interrupted during open (#1015)

## Java Iot Deps (com.microsoft.azure.sdk.iot:iot-deps:0.11.1)
- Increase dependency versions for commons-codec, jackson (#1039)

## Java Provisioning Device Client (com.microsoft.azure.sdk.iot.provisioning:provisioning-device-client:1.8.6)
- Increase dependency versions for commons-codec, jackson (#1039)

## Java Provisioning Service Client (com.microsoft.azure.sdk.iot.provisioning:provisioning-service-client:1.7.1)
- Increase dependency versions for commons-codec, jackson (#1039)

https://search.maven.org/artifact/com.microsoft.azure.sdk.iot/iot-service-client/1.27.0/jar
https://search.maven.org/artifact/com.microsoft.azure.sdk.iot/iot-device-client/1.29.0/jar
https://search.maven.org/artifact/com.microsoft.azure.sdk.iot/iot-deps/0.11.1/jar
https://search.maven.org/artifact/com.microsoft.azure.sdk.iot.provisioning/provisioning-device-client/1.8.6/jar
https://search.maven.org/artifact/com.microsoft.azure.sdk.iot.provisioning/provisioning-service-client/1.7.1/jar

old
{
    "device": "1.28.0",
    "service": "1.26.0",
    "deps": "0.11.0",
    "securityProvider": "1.4.0",
    "tpmEmulator": "1.1.1",
    "tpmHsm": "1.1.2",
    "diceEmulator": "1.1.1",
    "dice": "1.1.1",
    "x509": "1.1.4",
    "provisioningDevice": "1.8.5",
    "provisioningService": "1.7.0"
}

new
{
    "device": "1.29.0",
    "service": "1.27.0",
    "deps": "0.11.1",
    "securityProvider": "1.4.0",
    "tpmEmulator": "1.1.1",
    "tpmHsm": "1.1.2",
    "diceEmulator": "1.1.1",
    "dice": "1.1.1",
    "x509": "1.1.4",
    "provisioningDevice": "1.8.6",
    "provisioningService": "1.7.1"
}

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,11 +33,11 @@
...
         <x509-provider-artifact-id>x509-provider</x509-provider-artifact-id>
 
-        <iot-device-client-version>1.28.0</iot-device-client-version>
-        <iot-service-client-version>1.26.0</iot-service-client-version>
-        <iot-deps-version>0.11.0</iot-deps-version>
-        <provisioning-device-client-version>1.8.5</provisioning-device-client-version>
-        <provisioning-service-client-version>1.7.0</provisioning-service-client-version>
+        <iot-device-client-version>1.29.0</iot-device-client-version>
+        <iot-service-client-version>1.27.0</iot-service-client-version>
+        <iot-deps-version>0.11.1</iot-deps-version>
```
Comment (abhipsaMisra) [inline pom.xml:38]: q - is "deps" here similar to the .NET shared nuget:
- all other nuget packages have a dependency on shared
- if we release "shared", we have to release all the dependent nugets as well.

**Review by abhipsaMisra — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (timtay-microsoft) [reply pom.xml:38]: Yes. We also have lots of code refactoring in these libraries (prov-dev, prov-service) that may as well be released anyways

Comment (abhipsaMisra) [reply pom.xml:38]: won't we need to release the security provider and tpm jar as well then?

**Pull request merged**